Doc/library/test.rst

```diff
@@ -673,7 +1234,162 @@ The :mod:`test.support` module defines the following classes:
    On both platforms, the old value is restored by :meth:`__exit__`.
 
 
+.. class:: CleanImport(*module_names)
+
+   A context manager to force import to return a new module reference.  This
+   is useful for testing module-level behaviors, such as the emission of a
+   DeprecationWarning on import.  Example usage::
+
+      with CleanImport('foo'):
+          importlib.import_module('foo')  # New reference.
+
+
+.. class:: DirsOnSysPath(*paths)
+
+   A context manager to temporarily add directories to sys.path.
+
+   This makes a copy of :data:`sys.path`, appends any directories given
+   as positional arguments, then reverts :data:`sys.path` to the copied
+   settings when the context ends.
+
+   Note that *all* :data:`sys.path` modifications in the body of the
+   context manager, including replacement of the object,
+   will be reverted at the end of the block.
+
+
+.. class:: SaveSignals()
+
+   Class to save and restore signal handlers registered by the Python signal
+   handler.
+
+
+.. class:: Matcher()
```

Comment:
This class is used only in test_logging, and only with TestHandler. These classes should be documented together and with references to logging. I'm not sure they should be in `test.support`.